client/lib/site/utils.js

```diff
@@ -64,6 +64,10 @@ export default {
 		}
 	},
 
+	hasCustomDomain( site ) {
```

Comment:
What about sites with custom domain where it is not primary?

Cant it be detected from "site" object somehow? I think it has a lot of data